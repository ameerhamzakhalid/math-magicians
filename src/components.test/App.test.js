import React from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter as Router } from 'react-router-dom';
import App from '../components/App';

describe('App', () => {
  it('should render the Home component for the "/" route', () => {
    const { getByText } = render(
      <Router initialEntries={['/']}>
        <App />
      </Router>,
    );
    const homePage = getByText('Home');
    expect(homePage).toBeTruthy();
  });
});
