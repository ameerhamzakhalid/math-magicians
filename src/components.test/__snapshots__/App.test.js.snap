// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render the Home component for the "/" route 1`] = `
<nav
  class="nav"
>
  <h1>
    Math Magicians
  </h1>
  <ul>
    <li>
      <a
        aria-current="page"
        class="active"
        href="/"
      >
        Home
      </a>
    </li>
    |
    <li>
      <a
        class="none"
        href="/calculator"
      >
        Calculator
      </a>
    </li>
    |
    <li>
      <a
        class="none"
        href="/quotes"
      >
        Quotes
      </a>
    </li>
  </ul>
</nav>
`;
